controller/server.go

```diff
@@ -11,17 +12,48 @@ func (rw *ReadWrite) Config() config.Config {
 }
 
 func (rw *ReadWrite) Task(ctx context.Context, taskName string) (config.TaskConfig, error) {
+	// TODO handle ctx while waiting for driver lock if it is currently active
+	_, ok := rw.drivers.Get(taskName)
+	if !ok {
+		return config.TaskConfig{}, fmt.Errorf("a task with name '%s' does not exist or has not been initialized yet", taskName)
+	}
+
+	// TODO fill in return object
 	return config.TaskConfig{}, nil
 }
 
 func (rw *ReadWrite) TaskCreate(ctx context.Context, taskConfig config.TaskConfig) error {
-	return nil
+	_, err := rw.createTask(ctx, taskConfig)
+	return err
 }
 
 func (rw *ReadWrite) TaskCreateAndRun(ctx context.Context, taskConfig config.TaskConfig) error {
-	return nil
+	d, err := rw.createTask(ctx, taskConfig)
+	if err != nil {
+		return err
+	}
+
+	return rw.runTaskOnce(ctx, d)
 }
 
 func (rw *ReadWrite) TaskDelete(ctx context.Context, name string) error {
```

Comment:
All good, I see this was merged while I was going through it. No worries about this.

Reply:
Oh no sorry about that @wilkermichael ! Appreciate your time and review, especially since this is on the larger side. 
+1 to documenting public methods. I missed this and can do it the next PR i make.